pyproject.toml

```diff
@@ -0,0 +1,109 @@
+[tool.isort]
+profile = "black"
+line_length = 120
+known_first_party = ["neural_compressor", "neural_insights", "neural_solution"]
+extend_skip_glob = ["**/__init__.py"]
+
+
+[tool.black]
+line-length = 120
+
+
+[tool.codespell]
+skip = '*.po,*.ts,*.js,*.map,*.js.map,*.css.map,.azure-pipelines/scripts/codeScan/codespell/inc_dict.txt'
+count = ''
+quiet-level = 3
+ignore-words = ".azure-pipelines/scripts/codeScan/codespell/inc_dict.txt"
+
+
+[tool.ruff]
+# Exclude a variety of commonly ignored directories.
+exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".git-rewrite",
+    ".hg",
+    ".ipynb_checkpoints",
+    ".mypy_cache",
+    ".nox",
+    ".pants.d",
+    ".pyenv",
+    ".pytest_cache",
+    ".pytype",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    ".vscode",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "site-packages",
+    "venv",
+]
```

Comment:
```suggestion
```

ruff already excludes directories in gitignore by default https://docs.astral.sh/ruff/settings/#respect-gitignore

I suggest adding https://github.com/github/gitignore/blob/main/Python.gitignore to the current gitignore